src/overlays/actors/ovl_En_Horse_Ganon/z_en_horse_ganon.c

```diff
@@ -13,7 +13,27 @@ void EnHorseGanon_Destroy(EnHorseGanon* this, GlobalContext* globalCtx);
 void EnHorseGanon_Update(EnHorseGanon* this, GlobalContext* globalCtx);
 void EnHorseGanon_Draw(EnHorseGanon* this, GlobalContext* globalCtx);
 
-/*
+// internal functions
+void func_80A68660(Vec3s* data, s32 index, Vec3f* vec);
+void func_80A686A8(EnHorseGanon* this, GlobalContext* globalCtx);
+void func_80A68870(EnHorseGanon* this);
+void func_80A68AC4(EnHorseGanon* this);
+void func_80A68AF0(EnHorseGanon* this, GlobalContext* globalCtx);
+void func_80A68B20(EnHorseGanon* this);
+void func_80A68DB0(EnHorseGanon* this, GlobalContext* globalCtx);
+void func_80A68E14(EnHorseGanon* this, GlobalContext* globalCtx);
+void func_80A68FA8(EnHorseGanon* this, GlobalContext* globalCtx, ColliderJntSphItem* colliderSphereItem);
+
+// external functions
+void func_800A6888(GlobalContext*, s32*); // not exactly sure on 2nd arg type
+// not exactly sure on most of these
+void func_800A6330(Actor* this, GlobalContext* globalCtx, s32*,
+                   void(fn)(EnHorseGanon*, GlobalContext*, ColliderJntSphItem*), s32);
+
+// stolen from krim's z_skin branch, I edited the type of the 1st arg of the 1st function
+void func_800A6408(ColliderJntSphItem* arg0, s32 arg1, Vec3f* arg2, Vec3f* arg3);
+void func_800A663C(GlobalContext* globalCtx, UNK_PTR, SkeletonHeader*, AnimationHeader*);
```

Comment:
we have functions.h for prototypes of globally used functions. Id also generalize the type for the callback in func_800A6330 since this function would very likely be dealing with more than just EnHorseGanon*

Reply:
Given that `func_800A6330` seems to only be called by overlays involving horses that are ridden, and the coincidence that `EnHorseZelda` has the same size as `EnHorseGanon`, I wonder if `EnHorseGanon` might actually be some base struct for rideable horse overlays, meaning it would be correct here.

Also changing `func_80A68FA8` to take an `Actor*` instead of `EnHorseGanon*` and having a `EnHorseGanon* thisx` changes the stack size (`func_80A68FA8` is what gets passed to `func_800A6330`).

RE: moving prototypes to `functions.h`, 3/4 of them seem to have a struct from krim's branch as a parameter, would you still say I should move them to `functions.h`?

Reply:
The functions should be extern'd in `functions.h`, with the needed struct in `z64.h` for now.
There is a good chance that horse actors have a common struct for ridable actors, but anyway I would leave func_800A6330 as taking a `void*` for the `fn` argument, for now.

Also if you aren't sure about the type of an argument you can either not prototype the function for now, or use `UNK_TYPE` or another unk type. In general, avoid making prototypes if you aren't sure of them, unless it's necessary to match ofc.